### DISCUSSION TITLE
Could you please lower the bar of entrance for neorg?

### DISCUSSION THREAD

**Discussion by glyh** [nvim-neorg/neorg#1435] [General]
neovim is already niche for everyday devs, so it is even more so for looking upon managing plugins from user's perspective. 

I have been using neovim without trouble with any other plugins except `nvim-cmp` which is a mess for a while. But when using neorg, I feel like the specific way this plugin is organize enforces you to patching up stuff everywhere in order to make the life easier for the devs. And everything being strongly coupled together.

To use neorg...
- you have to use luarocks
- it works better with a specific package manager called `rocks.nvim`
- you can't lazy load it

I have tried neorg at some point, it works fine(I guess pre 8.0.0), with some copy pastes and modifying the Treesitter parsers config on my end but I can't make it work anyhow for now....

The default config on the README seems to be missing some features so that the tutorial video's view doesn't match mine as  the concealer is not enabled by default on my end for some reason, and I have to read 3~4 blogs to figure out what works. 

And then there's mess A, mess B... I'm not gonna elaborate as they are likely to be opened as issue, and I don't want to put much effort on them as an average user.

Why couldn't neorg be organized simpler and more portable, so the bar of entrance is lower?

**Comment by ALVAROPING1** [comment]
The requirement of luarocks is actually to make the plugin easier for both users and developers. By using it, dependencies can be managed by the plugin developers instead of the users, which allows them to be changed without the user having to change their config (which would be a breaking issue), and that allows developers to also rely more on already established libraries with extensive testing instead of having to remake everything from scratch and deal with all the bugs of a new implementation. On an ideal world, all plugins would use luarocks and neorg wouldn't be any different from other plugins. This change is unlikely to be reverted because as the plugin grows and more features are added, setting it up manually is going to have to become more and more difficult

The `rocks.nvim` is a package manager designed to use luarocks to manage plugins and dependencies instead of forcing the users to do it themselves. Unfortunately, `lazy` doesn't have any support for luarocks, but `luarocks.nvim` was created to add that support. You just need to specify that plugin as the only dependency of neorg if you are using `lazy` to get all the benefits of luarocks. Using this combination, neorg should work just as well as with `rocks.nvim`

Lastly, for lazy loading, ideally that should be done by the plugin itself as it can know what's important to setup directly and what can be lazy loaded (and it removes the burden of setting it up from the user), which as far as i'm aware is what neorg does. You can still lazy load it yourself through `lazy` though, the only issue i'm aware of with that is that you won't get `:Neorg` command completions because you didn't allow the plugin to set those up
